### DISCUSSION TITLE
How to get access to Meta real traces

### DISCUSSION THREAD

**Discussion by wangyuyue** [facebook/CacheLib#218] [Q&A]
Dear developers,
I want to test CacheLib using your real traces and analyze the distribution and changes of the traces.
But as I do `aws s3 ls s3://cachelib-workload-sharing/pub/kvcache/202206/`, it says `Unable to locate credentials. You can configure credentials by running "aws configure".`.

Does this mean the traces are not publically accessible yet? If not, how can I use it. And is there any other traces made public under the `cachelib-workload-sharing` directory? Thanks in advance.

**Comment by wangyuyue** [comment]
I think it's not any random `aws configure` that can access it.

**Reply by jaesoo-fb** [reply]
Hi @wangyuyue 

You don't need to sign-in. Instead just pass `--no-sign-request` for `aws-cli` commands.

**Reply by jaesoo-fb** [reply]
BTW, let us know if this does not work.

**Reply by wangyuyue** [reply]
Thanks, it works fine.
BTW, I notice that both generated workloads and the traces don't have a specified timestamp. So does it make sense to give a `zero` OpDelay time? If not, what would be a good time interval for the request's arrival?

**Reply by jaesoo-fb** [reply]
For QPS (Queries Per Second) control, we recommend to use the `opRatePerSec` (used by token bucket controller) which is set to 550K QPS by default.

**Reply by wangyuyue** [reply]
How about the OpDelayNs? Can I just make it zero? To be honest, I don't really understand the difference between OpDelayNs and opRatePerSec. It would be super helpful if you can briefly explain it. Thanks.

**Reply by jaesoo-fb** [reply]
Yep, just leave `OpDelayNs` to 0. Note that setting it along with the `opRatePerSec` will just make it hard to control the desired rate.

**Reply by wangyuyue** [reply]
And opRatePerSec controls the maximum rate, right? I mean if the CacheLib is unable to process so many queries in time, the input queries will be stalled instead of being discarded?

**Reply by jaesoo-fb** [reply]
Yes, right; `opRatePerSec` throttles the processing rate (as opposed to arrival rate) to be less than the max rate. If CacheLib is slow to process than the max rate, the processing of input queries will be stalled. There is no discard features.

See https://github.com/facebook/CacheLib/blob/main/cachelib/cachebench/runner/CacheStressor.h#L286 for reference; the `throttleFn` just adds some delays to slow down the processing.

**Comment by 1a1a11a** [comment]
Thank you for open-sourcing the traces! Where can I find more information about the traces (e.g., the time length of the trace, what type of the service it is)? Thank you！

**Reply by jaesoo-fb** [reply]
Hi @1a1a11a

You can find some limited information from [here](https://cachelib.org/docs/Cache_Library_User_Guides/Cachebench_FB_HW_eval#list-of-traces), where we will keep appending info for any new traces later.

**Reply by 1a1a11a** [reply]
Thank you! This is very helpful! One more question is the 930GB space dedicated to the workload, and is it still true that half of the SSD is used?

**Comment by wangyuyue** [comment]
Hi,
I find there is a configuration named `cacheDir` in the kvcache trace's configuration file. Could you give me some information about it? What effect does it have on the experiment? Thanks very much.

**Reply by wonglkd** [reply]
Probably this?
https://github.com/facebook/CacheLib/blob/fbf277ce6186fe6b438865af6bea0360e8ef3b92/cachelib/cachebench/cache/Cache.h#L108-L109

https://github.com/facebook/CacheLib/blob/fbf277ce6186fe6b438865af6bea0360e8ef3b92/cachelib/cachebench/util/CacheConfig.h#L68-L70

**Reply by wangyuyue** [reply]
Thanks, I should have read the code more carefully :)
